src/main/java/codeu/model/store/basic/ActivityStore.java

```diff
@@ -0,0 +1,167 @@
+        return new ArrayList<>(activities);
+    }
+
+    /**
+     * Access all Activity objects, sorted from oldest to most recent.
+     * Sorts all of the activities based on creation attribute.
+     */
+    public List<Activity> getAllSortedActivities() {
+        List<Activity> activitiesCopy = new ArrayList<>(activities);
+        activitiesCopy.sort(Collections.reverseOrder());
+        return activitiesCopy;
+    }
+
+    /**
+     * Access all Activity objects that were created on or after a specified date.
+     * 
+     * @param instant time that the Activity objects must be created on or after
+     */
+    public List<Activity> getActivitiesMadeOnOrAfter(Instant instant) {
+        int index = -1; // Index to take a sublist up to, exclusive
+        List<Activity> sortedActivities = getAllSortedActivities();
+        for (int i=0; i < sortedActivities.size(); i++) {
+            if (sortedActivities.get(i).getCreationTime().compareTo(instant) < 0) {
+                index = i;
+                break;
+            }
+        }
+        if (index != -1) {
+            return sortedActivities.subList(0,index);
+        } else {
+            return sortedActivities;
+        }
+    }
+
+    /**
+     * Access a specified number of Activity objects from a specified index
+     * 
+     * @param startIndex index to begin subarray
+     * @param batchSize number of Activity objects to access
+     */
+    public List<Activity> getActivitiesBatch(int startIndex, int batchSize) {
+        if (startIndex + batchSize > activities.size()) {
+            return getAllSortedActivities().subList(startIndex, activities.size());
+        } else {
+            return getAllSortedActivities().subList(startIndex, startIndex + batchSize);
+        }
+    }
+
+    /** Add a new activity to the current set of activities known to the application. */
+    public void addActivity(Activity activity) {
+        activities.add(activity);
+        persistentStorageAgent.writeThrough(activity);
+    }
+
+    public void addActivity(User user) {
+        String[] displayStringParameters = {user.getCreationTime().toString(),
+                                            user.getName()}; 
+        Activity activity = new Activity(Activity.ActivityType.UserRegistered, 
+                                         user.getId(),
+                                         user.getCreationTime(),
+                                         displayStringParameters);
+        activities.add(activity);
+        persistentStorageAgent.writeThrough(activity);
+    }
+
+    public void addActivity(Message message) {
+        String[] displayStringParameters = {message.getCreationTime().toString(),
+                                            UserStore.getInstance()
+                                                .getUser(message.getAuthorId())
+                                                .getName(),
+                                            ConversationStore.getInstance()
```

Comment:
Most times we are adding an Message activity we know the conversationTitle so maybe just pass it in as a parameter.